presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -408,6 +410,38 @@ protected static CacheQuota generateCacheQuota(HiveSplit hiveSplit)
 
         Optional<BucketAdaptation> bucketAdaptation = bucketConversion.map(conversion -> toBucketAdaptation(conversion, regularAndInterimColumnMappings, tableBucketNumber, ColumnMapping::getIndex));
 
+        if (isHudiEnabled(session) && shouldUseRecordReaderFromInputFormat(configuration, storage, customSplitInfo)) {
+            return getPageSourceFromCursorProvider(
```

Comment:
This method can return Optional.empty(), Is that an acceptable return value ?

Reply:
Yes, that was the behavior even before. If the page source is empty then we throw exception in the caller:
```
if (pageSource.isPresent()) {
    return pageSource.get();
}
throw new IllegalStateException("Could not find a file reader for split " + hiveSplit);
```